internal/command/state_show.go

```diff
@@ -162,16 +162,24 @@ func (c *StateShowCommand) Run(args []string) int {
 
 	// check if the resource has a configured provider, otherwise this will use the default provider
 	rs := state.Resource(addr.ContainingResource())
-	absPc := addrs.AbsProviderConfig{
+	absResourceProvider := addrs.AbsProviderConfig{
 		Provider: rs.ProviderConfig.Provider,
 		Alias:    rs.ProviderConfig.Alias,
 		Module:   addrs.RootModule,
 	}
+
+	absInstanceProvider := addrs.AbsProviderConfig{
```

Comment:
We need to test this change thoroughly. 
I didn't understand why the `absResourceProvider` has its Module set to `addrs.RootModule`. I took the same approach with `absInstanceProvider`, but we should validate this is correct.